src/app.ts

```diff
@@ -13,14 +13,53 @@ interface TemperatureSummary {
   average: number
 }
 
+const records: Array<TemperatureReading> = []
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  records.push(...readings)
+  records.sort((a, b) => a.time.getTime() - b.time.getTime())
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const EMPTY_ARRAY = 0
+
+  //Filter elements by date and city
+  const recordsOrderByCityAndDate: TemperatureReading[] = records.filter(
+    (item) => item.time.getDate() === date.getDate() && item.city === city,
+  )
+
+  //Order records from high to lower temperature
+  const recordsOrderByTemp = [...recordsOrderByCityAndDate].sort(
+    (a, b) => b.temperature - a.temperature,
+  )
+
+  // If no exist any data return null
+  if (recordsOrderByCityAndDate.length === EMPTY_ARRAY) return null
+
+  enum ArrayProperties {
+    LENGTH_ARRAY = recordsOrderByCityAndDate.length,
+    FIRST_ITEM = 0,
+    LAST_ITEM = LENGTH_ARRAY - 1,
+  }
```

Comment:
we could move this enum outside our function, like an interface, type, class, etc.